File: 0083/0083.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/1/16 19:17
# File: 0083.py
# Desc: 

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if None == head: return head
        doneDic = set()
        ret = ListNode(head.val)
        cur = ret
        doneDic.add(ret.val)
        while head != None:
            if head.val not in doneDic:
                doneDic.add(head.val)
                cur.next = ListNode(head.val)
                cur = cur.next
            head = head.next
            
        return ret

File: 0083/0083.java
// Author: washing
// DateTime: 2021/3/26 11:59
// File: 0083.java
// Desc: 

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) 
            return head;
        int num = head.val;
        ListNode ret = new ListNode(num);
        ListNode cur = ret;
        while (head.next != null){
            head = head.next;
            int v = head.val;
            if (num != v){
                num = v;
                cur.next = new ListNode(num);
                cur = cur.next;
            }
        }
        return ret;
    }
}